### ISSUE TITLE
page.search_for is case sensitive for accented chars

### ISSUE THREAD

**Issue by jrom99** [pymupdf/PyMuPDF#1303]
## Describe the bug (mandatory)
Searching for accented chars (ó, é, ô...) is case sensitive.

## To Reproduce (mandatory)
```
# To create pdf
import fitz
doc = fitz.open()
doc.insert_page(0)
page = doc[0]
page.insert_text(fitz.Point(100, 50, "genótipo")
page.insert_text(fitz.Point(100, 100), "GENÓTIPO")
page.insert_text(fitz.Point(100, 150), "pétala")
page.insert_text(fitz.Point(100, 200), "PÉTALA")
doc.save("test.pdf")

# To interpret pdf
import fitz
doc = fitz.open("test.pdf")
page = doc[0]
page.search_for("é")  # gives only one result
page.search_for("pétala")  # gives only one result
page.search_for("tala")  # gives two results
page.search_for("pÉtala")  # gives one result
page.search_for("PÉTALA") # gives same result

# to highlight pdf
for rect in page.search_for("genótipo"):
    page.draw_rect(rect, color=(1,0,0))

for rect in page.search_for("GENÓTIPO"):
    page.draw_rect(rect, color=(0,0,1))

doc.save("out.pdf")
quit()
```

## Expected behavior (optional)
Searching for "genótipo" or "pétala" should give two results (be case insensitive).

## Screenshots (optional)
![Highlighted words "genótipo" in output pdf file, uppercase in blue, lowercase in red](https://i.imgur.com/B3T8cO5.png)

## Your configuration (mandatory)
3.9.5 (default, May 11 2021, 08:20:37) 
[GCC 10.3.0] 
 linux 
 
PyMuPDF 1.18.19: Python bindings for the MuPDF 1.18.0 library.
Version date: 2021-09-16 16:45:29.
Built for Python 3.9 on linux (64-bit).

PyMuPDF was installed using pip, my OS is Ubuntu 21.04

**Comment by JorjMcKie** [comment]
> Searching for "genótipo" or "pétala" should give two results (be case insensitive).

No: this works as documented. Case insensitivity does not work outside the ASCII range.

**Comment by jrom99** [comment]
Thanks for the information, would it be possible to add it as a feature?

**Comment by JorjMcKie** [comment]
> 
> 
> Thanks for the information, would it be possible to add it as a feature?

I am afraid not: this is not my code, but original MuPDF's. I know they have marked the relevant source with a respective "TODO" ... but who knows. The problem is that most of the different variants of e.g. "a" are coded completely differently - it is not simply a matter of one bit being on or off as it is the case for "a" and "A". Just have a look at this:
```python
>>> bin(ord("a"))
'0b1100001'
>>> bin(ord("A"))
'0b1000001'
>>> bin(ord("ä"))
'0b11100100'
>>> bin(ord("Ä"))
'0b11000100'
>>> bin(ord("á"))
'0b11100001'
>>> bin(ord("à"))
'0b11100000'
>>> bin(ord("â"))
'0b11100010'
>>> bin(ord("Â"))
'0b11000010'
>>> bin(ord("Á"))
'0b11000001'
>>> bin(ord("À"))
'0b11000000'
>>> # now imagine French cedille, Scandinavian and Slavian forms and what not ...
```

**Comment by jrom99** [comment]
Thanks again, this seems like a thorny issue. It would be great though if MuPDF could search using a `str.casefold()` like method, better yet a `unicodedata.normalize("NFC", char)`
